src/Kobo/ImageProcessor/CoverTransformer.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace App\Kobo\ImageProcessor;
+
+class CoverTransformer
+{
+    public function __construct()
+    {
+    }
+
+    private function fallback(string $coverPath): void
+    {
+        readfile($coverPath);
+    }
+
+    public function streamFile(string $coverPath, int $maxWidth, int $maxHeight, bool $grayscale = false): void
+    {
+        // Check if GD extension is loaded
+        if (false === function_exists('imagecreatetruecolor')) {
+            $this->fallback($coverPath);
+
+            return;
+        }
+
+        // We only support jpeg & png & gif
+        if (false === str_ends_with($coverPath, '.jpg')
+            && false === str_ends_with($coverPath, '.png')
+            && false === str_ends_with($coverPath, '.gif')
+            && false === str_ends_with($coverPath, '.jpeg')
+        ) {
+            $this->fallback($coverPath);
+
+            return;
+        }
+
+        // Resize image based on the width and height
+        $sizes = getimagesize($coverPath);
+        if ($sizes === false) {
+            $this->fallback($coverPath);
+
+            return;
+        }
+        list($originalWidth, $originalHeight) = $sizes;
+
```

Comment:
why not use imaginebundle?

Reply:
We can discuss this, but it seems complicated to me.